server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -397,6 +398,47 @@ public Metadata withCoordinationMetadata(CoordinationMetadata coordinationMetada
         );
     }
 
+    /**
+     * Creates a copy of this instance updated with the given {@link IndexMetadata} that must only contain changes to primary terms
+     * and in-sync allocation ids relative to the existing entries. This method is only used by
+     * {@link org.elasticsearch.cluster.routing.allocation.IndexMetadataUpdater#applyChanges(Metadata, RoutingTable)}.
+     * @param updates map of index name to {@link IndexMetadata}.
+     * @return updated metadata instance
+     */
+    public Metadata withAllocationAndTermUpdatesOnly(Map<String, IndexMetadata> updates) {
```

Comment:
This sounds very specific. Would it make sense to have it as a `private static Metadata withAllocationAndTermUpdatesOnly(Metadata original, Map<String, IndexMetadata> updates)` in `IndexMetadataUpdater` so nobody misuses it somewhere else?

Reply:
I agree, it's a little messy. The problem is, we don't really get anything out of moving to this kind of private method because then we'd have to expose the raw constructor of `Metadata` to do the work elsewhere wouldn't we?